stratum/hal/lib/bcm/bcm_sdklt_config_util.h

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright 2019-present Open Networking Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef STRATUM_HAL_LIB_BCM_BCM_SDK_CONFIG_UTIL_H_
+#define STRATUM_HAL_LIB_BCM_BCM_SDK_CONFIG_UTIL_H_
+
+#include <string>
+
+#include "yaml-cpp/yaml.h"
+#include "absl/container/flat_hash_map.h"
+#include "stratum/hal/lib/bcm/bcm.pb.h"
+#include "stratum/hal/lib/bcm/utils.h"
+#include "stratum/glue/status/statusor.h"
+#include "stratum/glue/integral_types.h"
+#include "stratum/lib/constants.h"
+#include "stratum/lib/macros.h"
+#include "stratum/glue/status/status_macros.h"
+
+namespace stratum {
+namespace hal {
+namespace bcm {
+
+constexpr int kMaxSerdesLaneNumber = 15;
+
+// Data structure to hold lane settings for BCM port macro
+struct SerdesLaneSetting {
+  uint32 speed_mbps[16];
+  uint32 lane_map[16];
+  uint64 tx_lane_map;
+  uint64 rx_lane_map;
+  uint64 rx_polarity_flip;
+  uint64 tx_polarity_flip;
+  BcmPort_OpMode op_mode[4];
+};
+
+const std::string kSdkltOpModeDefault = "PC_PM_OPMODE_DEFAULT";
+const std::string kSdkltOpModeQsgmii = "PC_PM_OPMODE_QSGMII";
+const std::string kSdkltOpModeGphy = "PC_PM_OPMODE_GPHY";
+
+::util::StatusOr<std::string> toBcmSdkltOpModeStr(BcmPort_OpMode op_mode) {
+  switch (op_mode) {
+    case BcmPort_OpMode_OPMODE_DEFAULT:
+      return kSdkltOpModeDefault;
+    case BcmPort_OpMode_OPMODE_QSGMII:
+      return kSdkltOpModeQsgmii;
+    case BcmPort_OpMode_OPMODE_GPHY:
+      return kSdkltOpModeGphy;
+    default:
+      RETURN_ERROR(ERR_INVALID_PARAM) << "Unknown operation mode " << op_mode;
+  }
+};
```

Comment:
Would it make sense to move this into a protobuf? bcm.proto as enum?

Reply:
I believe that bcm.proto is a general structure for both SDKLT and the private SDK.
And `PC_PM_...` is for SDKLT only. (maybe I am wrong?)

Reply:
You are correct. The config generator obviously depends on the SDK.